### ISSUE TITLE
False refactoring suggestion 'Call to static member via a derived class' RECS0030

### ISSUE THREAD

**Issue by usr-sse2** [mono/monodevelop#7081]
**Version Used**: 
Visual Studio for Mac 2019 Preview 2

**Steps to Reproduce**:
```cs
using DBG = System.Diagnostics.Debugger;
namespace Common
{
    public static class Debugger
    {
        public static void SoftBreak()
        {
            if (DBG.IsAttached)
                DBG.Break();
        }
    }
}
```

**Expected Behavior**:
No warnings

**Actual Behavior**:
On both lines a warning "Call to static member via a derived class" and a fix suggestion "Use base qualifier 'DBG'" which replaces 'DBG' with 'DBG', so, does nothing.

> VS bug [#787966](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/787966)

**Comment by aaronfranke** [comment]
I'm also experiencing this on MonoDevelop 7.7 build 1869. I have:

    using real_t = System.Single;

And my code with the false warning looks like this:

![warning](https://user-images.githubusercontent.com/1646875/52514170-89223d00-2bdd-11e9-9f9b-ad67c42378a4.png)

**Comment by Therzok** [comment]
https://github.com/mono/RefactoringEssentials/blob/roslyn-3/RefactoringEssentials/CSharp/Diagnostics/Synced/PracticesAndImprovements/AccessToStaticMemberViaDerivedTypeAnalyzer.cs#L77

We might need to lookup typeinfo in some situations to see if it's an alias.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-6] Comment by **xamarinc**



Fixed in version 8.1.0.1764 (master-vnext)



Author: therzok  
 Commit: 8792c806ee9e3c4e1cd0f6f93fb166c6aab8c107 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-8] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Author: therzok  
 Commit: 8792c806ee9e3c4e1cd0f6f93fb166c6aab8c107 (xamarin/monodevelop)
